docs/config.md

```diff
@@ -1872,6 +1913,203 @@ Default: `16`
+periodically refresh the `KeyStore` at each [`Provide.ReprovideInterval`](#providereprovideinterval)
+by providing it a channel of all the keys it is expected to contain according
+to the [`Provide.Strategy`](#providestrategy). During this operation,
+all keys in the `Keystore` are purged, and only the given ones remain scheduled.
+
+#### `Provide.Sweep.Enabled`
+
+Whether Provide Sweep is enabled. If not enabled, the
+[`boxo/provider`](https://github.com/ipfs/boxo/tree/main/provider) is used for
+both provides and reprovides.
+
+> [!NOTE]
+> This feature is opt-in for now, but will become the default in a future release.
+> Eventually, this configuration flag will be removed once the feature is stable.
+
+Default: `false`
+
+Type: `flag`
+
+#### `Provide.Sweep.MaxWorkers`
+
+The maximum number of workers used by the `SweepingReprovider` to provide and
+reprovide CIDs to the DHT swarm.
+
+A worker performs Kademlia `GetClosestPeers` operations (max 1 at a time) to
+explore a region of the DHT keyspace, and then sends provider records to the
+nodes from that keyspace region. `GetClosestPeers` is capped to `10` concurrent
+connections [`amino` DHT
+defaults](https://github.com/libp2p/go-libp2p-kad-dht/blob/master/amino/defaults.go).
+The number of simultaneous connections used to send provider records is defined
+by
+[`Provide.Sweep.MaxProvideConnsPerWorker`](#providesweepMaxProvideConnsPerWorker).
+
+The workers are split between two tasks categories:
+
+1. Periodic reprovides (see
+   [`Provide.Sweep.DedicatedPeriodicWorkers`](#providesweepdedicatedperiodicworkers))
+2. Burst provides (see
+   [`Provide.Sweep.DedicatedBurstWorkers`](#providesweepdedicatedburstworkers))
+
+[`Provide.Sweep.DedicatedPeriodicWorkers`](#providesweepdedicatedperiodicworkers)
+workers are allocated to the periodic reprovides only,
+[`Provide.Sweep.DedicatedBurstWorkers`](#providesweepdedicatedburstworkers)
+workers are allocated to burst provides only, and the rest of
+[`Provide.Sweep.MaxWorkers`](#providesweepmaxworkers) can be used for
+either task (first come, first served).
+
+Default: `4`
+
+Type: `optionalInteger` (non-negative)
+
+#### `Provide.Sweep.DedicatedPeriodicWorkers`
+
+Number of workers dedicated to periodic keyspace region reprovides.
+
+Among the [`Provide.Sweep.MaxWorkers`](#providesweepmaxworkers), this
+number of workers will be dedicated to the periodic region reprovide only. In
+addition to these, if there are available workers in the pool, they can also be
+used for periodic reprovides.
+
+Default: `2`
+
+Type: `optionalInteger` (`0` means there are no dedicated workers, but the
+operation can be performed by free non-dedicated workers)
+
+#### `Provide.Sweep.DedicatedBurstWorkers`
+
+Number of workers dedicated to burst provides.
+
+Burst provides are triggered when new keys must be advertised to the DHT
+immediately, or when a node comes back online and must catch up the reprovides
+that should have happened while it was offline.
```

Comment:
@guillaumemichel is Burst worker pool used for ad-hoc `ipfs routing provide --recursive=true|false $cid` or `ipfs provide once --recursive=true|false $cid`?

Reply:
It is used for `SweepingProvider` operations that need to happen ASAP:
1. New provides (`ipfs routing provide start --recursive=true|false $cid`, `ipfs routing provide once --recursive=true|false $cid`, and blocks fetched from bitswap/http matching `Provide.Strategy`
2. Catching up with reprovides after being disconnected/offline for a while.

The reason to have 2 types of workers (each with reserved seats), is that Burst work (e.g adding a massive number of CIDs, or reconnecting to network), shouldn't delay the reprovide operation, and vice-versa.